### PR TITLE
Fixes race condition in notification.go

### PR DESCRIPTION
**Pull request by koox00**
#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: app/notification.go [hunk 1]
```diff
@@ -258,7 +260,7 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 				// add new followers to existing followers
 				if threadMembership.Following && !followers.Contains(userID) {
```
Comment (agnivade) [inline app/notification.go:261]: Please expand the mutex critical section to include the entire if condition. That should fix the race. I don't believe there is a need to introduce an additional map.

**Review by agnivade — APPROVED** [review]

**Review by agnivade — APPROVED** [review]
Thanks!

**Review by jwilander — APPROVED** [review]

**Comment by amyblais** [comment]
@koox00 Is this for v6.0?

**Comment by koox00** [comment]
> @koox00 Is this for v6.0?

Yes

**Pull request merged**